jbi/actions/default.py

```diff
@@ -95,15 +97,30 @@ def __init__(self, steps: dict[Operation, list[Callable]], **parameters):
 
     def __call__(self, context: ActionContext) -> ActionResult:
         """Called from `runner` when the action is used."""
+        has_produced_request = False
+
         for step in self.steps[context.operation]:
             context = context.update(current_step=step.__name__)
             try:
                 context = step(context=context, **self.parameters)
             except IncompleteStepError as exc:
                 # Step did not execute all its operations.
                 context = exc.context
+                statsd.incr(
+                    f"jbi.action.{context.jira.project.lower()}.incomplete.count"
+                )
+            except Exception:
+                if has_produced_request:
+                    # Count the number of workflows that produced at least one request,
+                    # but could not complete entirely with success.
+                    statsd.incr(
+                        f"jbi.action.{context.jira.project.lower()}.aborted.count"
+                    )
+                raise
 
             step_responses = context.responses_by_step[step.__name__]
+            if step_responses:
+                has_produced_request = True
```

Comment:
I'm having a hard time seeing how this `has_produced_request` variable works to determine if a workflow was aborted or not. What does this signal vs just incrementing when we hit an `Exception`?

Reply:
I was thinking of this:
- if a workflow crashes before or during its first request, nothing was executed on the Jira side
- if a workflow crashes after a request was executed, the workflow was partially executed, "aborted", and the state on the Jira side is potentially inconsistent
- if we would count exceptions it's more or less equivalent to counting 5XX responses

Reply:
Ahh okay, thanks for that clarification